single arg ok for primerange

smichr opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

close #19118 

#### Brief description of what is fixed or changed

give `primerange` a single-arg option and document the use of `prime` to reference by index the desired number of primes to generate
```python
>>> primerange(prime(10))  # up to 10th
[2, 3, 5, 7, 11, 13, 17, 19, 23]
>>> primerange(prime(10)+1)  # including the 10th
[2, 3, 5, 7, 11, 13, 17, 19, 23, 29]
```
#### Other comments


#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* ntheory
  * `primerange` now accepts a single argument, a,  which gives the range [2, a)
<!-- END RELEASE NOTES -->

smichr committed:
single arg ok for primerange

oscarbenjamin commented on sympy/ntheory/generate.py:
```diff
@@ -692,6 +696,7 @@ def primerange(a, b):
         ========
 
+        prime : Return the ith prime
         nextprime : Return the ith prime greater than n
```
What does `ith prime greater than n` mean?

smichr replied:
It should read "return the next prime greater than i"

smichr committed:
annotate and unify docstrings

smichr commented on sympy/ntheory/generate.py:
```diff
@@ -692,6 +728,7 @@ def primerange(a, b):
         See Also
         ========
 
+        prime : Return the ith prime
```
```suggestion
        prime : Return the nth prime
```

smichr committed:
Update sympy/ntheory/generate.py

use `nth` instead of `ith` to match function arg and to avoid confusion with the next line of docstring

smichr commented:
This adds functionality to function `primerange` that was already agreed upon for the sieve method with the same name.
